generators/python-v2/codegen/src/ast/TypeInstantiation.ts

```diff
@@ -90,27 +91,40 @@ export class TypeInstantiation extends AstNode {
     }
 
     public static list(values: AstNode[]): TypeInstantiation {
-        return new this({ type: "list", values });
+        const list = new this({ type: "list", values });
+        values.forEach((value) => list.inheritReferences(value));
+        return list;
```

Comment:
We perform similar reference tracking in `python.Type` and failed to include it here in `python.TypeInstantiation`.